packages/core/core/src/TransformerRunner.js

```diff
@@ -38,8 +41,12 @@ type Opts = {|
 
 type GenerateFunc = (input: IMutableAsset) => Promise<GenerateOutput>;
 
+const CORE_VERSION = corePackage.version;
 const BUFFER_LIMIT = 5000000; // 5mb
 
+// These keys can affect asset output, so if they differ, we should always transform
+const IMPACTFUL_OPTIONS = ['minify', 'hmr', 'scopeHoist'];
```

Comment:
Does scopehoisting bust the complete cache? Because with a typescript project, the typescript-transform could be reused (and css/html/... as well).

Reply:
I think this probably falls into letting transformers decide if they'd reuse cached results. @padmaia has been thinking about how to get transformers participating in cache invalidation. 

For now I think leaning on being as conservative as possible is probably the way to go. But right now we still don't invalidate if a plugin updates, which seems like a big oversight. I think @padmaia's working on that too.